src/group_theory/nilpotent.lean

```diff
@@ -690,6 +691,49 @@ begin
   exact upper_central_series_eq_top_iff_nilpotency_class_le.mpr h,
 end
 
+section prod
+
+variables {G₁ : Type*} {G₂ : Type*} [group G₁] [group G₂]
+
+@[simp]
+lemma lower_central_series_prod (n : ℕ):
+  lower_central_series (G₁ × G₂) n = (lower_central_series G₁ n).prod (lower_central_series G₂ n) :=
+begin
+  induction n with n ih,
+  { simp, },
+  { calc lower_central_series (G₁ × G₂) n.succ
+        = ⁅lower_central_series (G₁ × G₂) n, ⊤⁆  : rfl
+    ... = ⁅(lower_central_series G₁ n).prod (lower_central_series G₂ n), ⊤⁆ : by rw ih
+    ... = ⁅(lower_central_series G₁ n).prod (lower_central_series G₂ n), (⊤ : subgroup G₁).prod ⊤⁆ :
+      by simp
+    ... = ⁅lower_central_series G₁ n, (⊤ : subgroup G₁)⁆.prod ⁅lower_central_series G₂ n, ⊤⁆ :
+      general_commutator_prod_prod _ _ _ _
+    ... = (lower_central_series G₁ n.succ).prod (lower_central_series G₂ n.succ) : rfl }
+end
+
+/-- Products of nilpotent groups are nilpotent -/
+instance is_nilpotent_prod [is_nilpotent G₁] [is_nilpotent G₂] :
+  is_nilpotent (G₁ × G₂) :=
+begin
+  rw nilpotent_iff_lower_central_series,
+  refine ⟨max (group.nilpotency_class G₁) (group.nilpotency_class G₂), _ ⟩,
+  rw lower_central_series_prod,
+  rw lower_central_series_eq_bot_iff_nilpotency_class_le.mpr (le_max_left _ _),
+  rw lower_central_series_eq_bot_iff_nilpotency_class_le.mpr (le_max_right _ _),
+  rw bot_prod_bot,
```

Comment:
It hardly matters here because the names are so long that we still have to break lines but our preferred style is:
```suggestion
  rw [lower_central_series_prod,
    lower_central_series_eq_bot_iff_nilpotency_class_le.mpr (le_max_left _ _),
    lower_central_series_eq_bot_iff_nilpotency_class_le.mpr (le_max_right _ _),
    bot_prod_bot],
```
(This makes me wonder if some of these lemmas should use the abbreviation `lcs` for `lower_central_series`.)